### ISSUE TITLE
rieman-tools not installable 

### ISSUE THREAD

**Issue by wfranzini** [NixOS/nixpkgs#41913]
## Issue description

rieman-tools fails to  install on nixos-18.03 due to a dependency problem with yajl-ruby

the problem seems to be the same as 
https://github.com/brianmario/yajl-ruby/issues/164

where it's sttaed that yajl-ruby 1.2.x requires ruby-2.3

### Steps to reproduce

nix-env -iA nixos.riemann-tools
...
```
installing 'riemann-tools-0.2.6'
these derivations will be built:
  /nix/store/a1s5y6ik08fmiry2l42vh699wxzqiwmz-ruby2.4.4-yajl-ruby-1.2.1.drv
  /nix/store/ipa9gz1lgvs324ffinggrpcmx6fzlrca-ruby2.4.4-riemann-tools-0.2.6.drv
  /nix/store/w2bcrqag20b1x7ml7lb92qqxvjy1aq5s-riemann-tools-0.2.6.drv
building '/nix/store/a1s5y6ik08fmiry2l42vh699wxzqiwmz-ruby2.4.4-yajl-ruby-1.2.1.drv'...
unpacking sources
patching sources
configuring
no configure script, doing nothing
installing
buildFlags:
WARNING:  You build with buildroot.
  Build root: /
  Bin dir: /nix/store/9s3dl1ip6naznncr554xa6sgaikvpnpz-ruby2.4.4-yajl-ruby-1.2.1/lib/ruby/gems/2.4.0/bin
  Gem home: /nix/store/9s3dl1ip6naznncr554xa6sgaikvpnpz-ruby2.4.4-yajl-ruby-1.2.1/lib/ruby/gems/2.4.0
Building native extensions. This could take a while...
ERROR:  Error installing /nix/store/i8jpc3y9zlk84wm2jyszjl9a9xqkn9xs-yajl-ruby-1.2.1.gem:
        ERROR: Failed to build gem native extension.

    current directory: /nix/store/9s3dl1ip6naznncr554xa6sgaikvpnpz-ruby2.4.4-yajl-ruby-1.2.1/lib/ruby/gems/2.4.0/gems/yajl-ruby-1.2.1/ext/yajl
/nix/store/x932mdkafx2c4dzqznj7cwa6974lywr0-ruby-2.4.4/bin/ruby -r ./siteconf20180613-6462-v2052b.rb extconf.rb
creating Makefile

current directory: /nix/store/9s3dl1ip6naznncr554xa6sgaikvpnpz-ruby2.4.4-yajl-ruby-1.2.1/lib/ruby/gems/2.4.0/gems/yajl-ruby-1.2.1/ext/yajl
make "DESTDIR=" clean

current directory: /nix/store/9s3dl1ip6naznncr554xa6sgaikvpnpz-ruby2.4.4-yajl-ruby-1.2.1/lib/ruby/gems/2.4.0/gems/yajl-ruby-1.2.1/ext/yajl
make "DESTDIR="
compiling yajl.c
compiling yajl_alloc.c
compiling yajl_buf.c
compiling yajl_encode.c
compiling yajl_ext.c
yajl_ext.c: In function 'rb_yajl_parser_parse':
yajl_ext.c:471:17: warning: variable 'stat' set but not used [-Wunused-but-set-variable]
     yajl_status stat;
                 ^~~~
yajl_ext.c: In function 'rb_yajl_encoder_enable_json_gem_ext':
yajl_ext.c:881:22: error: 'rb_cFixnum' undeclared (first use in this function); did you mean 'rb_isalnum'?
     rb_define_method(rb_cFixnum, "to_json", rb_yajl_json_ext_fixnum_to_json, -1);
                      ^~~~~~~~~~
                      rb_isalnum
yajl_ext.c:881:22: note: each undeclared identifier is reported only once for each function it appears in
yajl_ext.c: At top level:
cc1: warning: unrecognized command line option '-Wno-self-assign'
cc1: warning: unrecognized command line option '-Wno-constant-logical-operand'
cc1: warning: unrecognized command line option '-Wno-parentheses-equality'
make: *** [Makefile:242: yajl_ext.o] Error 1

make failed, exit code 2

Gem files will remain installed in /nix/store/9s3dl1ip6naznncr554xa6sgaikvpnpz-ruby2.4.4-yajl-ruby-1.2.1/lib/ruby/gems/2.4.0/gems/yajl-ruby-1.2.1 for inspection.
Results logged to /nix/store/9s3dl1ip6naznncr554xa6sgaikvpnpz-ruby2.4.4-yajl-ruby-1.2.1/lib/ruby/gems/2.4.0/extensions/x86_64-linux/2.4.0/yajl-ruby-1.2.1/gem_make.out
builder for '/nix/store/a1s5y6ik08fmiry2l42vh699wxzqiwmz-ruby2.4.4-yajl-ruby-1.2.1.drv' failed with exit code 1
cannot build derivation '/nix/store/w2bcrqag20b1x7ml7lb92qqxvjy1aq5s-riemann-tools-0.2.6.drv': 1 dependencies couldn't be built
error: build of '/nix/store/w2bcrqag20b1x7ml7lb92qqxvjy1aq5s-riemann-tools-0.2.6.drv' failed
```

## Technical details

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the
results.

```
 - system: `"x86_64-linux"`
 - host os: `Linux 4.14.48, NixOS, 18.03.132645.43c77db3aa5 (Impala)`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.0.4`
 - channels(root): `"nixos-18.03.132645.43c77db3aa5"`
 - channels(walter): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos/nixpkgs`
```

**Comment by manveru** [comment]
You can try my PR: https://github.com/NixOS/nixpkgs/pull/41927

**Comment by xeji** [comment]
Fixed by #41927, also backported to 18.03
